File: 프로그래머스/unrated/120812. 최빈값 구하기/최빈값 구하기.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<int> array)
{
    unordered_map<int, int> frequency;

    for (int num : array) 
        frequency[num]++;

    int max_freq = 0;
    int mode = -1;

    for (const auto& kv : frequency) 
    {
        if (kv.second > max_freq) 
        {
            max_freq = kv.second;
            mode = kv.first;
        }
    }

    int count = 0;
    for (const auto& kv : frequency) 
    {
        if (kv.second == max_freq) 
        {
            count++;
            if (count > 1)
                return -1;  
        }
    }

    return mode;  
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    const frequency = {};
    let maxFreq = 0;
    let mode = -1;
    
    array.forEach(num => {
        if (frequency[num])
            frequency[num]++;
        else
            frequency[num] = 1;
        
        if (frequency[num] > maxFreq) {
            maxFreq = frequency[num];
            mode = num;
		}
    });
    
    const modes = Object.values(frequency).filter(freq => freq === maxFreq);
    if (modes.length > 1)
        return -1;
    
    return mode;
}
